### PR TITLE
Feat/authentication

### LINKED ISSUE #2: Functional: Registration and Login
**User Story**: As a User, I want to be able to create a new account or log in to an existing one so that I can securely access my profile.

**Confirmation**: Users can create or log in to already created accounts.

### PR DESCRIPTION
**Pull request by hdprajwal**
This pull request has the following changes to the backend.

### Dependency Updates:
* Removed `spring-boot-starter-security` and added `lombok` dependency. 

### New Model and Entity Classes:
* Added a new `User` model class with Lombok annotations.
* Added a new `UserEntity` class with JPA annotations.
* Added a new `UserEntityId` class for composite key handling.

### New Controller and Service Classes:
* Added a new `AuthController` class for handling authentication endpoints.
* Implemented `AuthService` in a new `AuthServiceImpl` class.

### New DTO and Handler Classes:
* Added a new `AuthResponse` DTO class.
* Added a new `LoginRequest` DTO class.
* Added a new `RegisterRequest` DTO class.
* Added a new `ResponseHandler` class for generating response entities.

### CODE REVIEW

**Review by ChiragBellara7 — COMMENTED** [review]
The column names in the table are "profile_picture", "logged_in_status" while their associated field names in the UserEntity file are profilePicture, and loggedInStatus.

Would that work? Or is this a mistake?

**Comment by hdprajwal** [comment]
Both the naming convention seem to work, But it would be a point of confussion if i leave it like this,
I will update the code to follow the same naming convention as that of the database tables.

File: mainframe/src/main/java/com/PiXl/mainframe/entities/UserEntity.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+@Entity()
+@Table(name = "user")
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor
+@ToString
+public class UserEntity {
+    @EmbeddedId
+    private UserEntityId id;
```
Comment (ChiragBellara7) [inline mainframe/src/main/java/com/PiXl/mainframe/entities/UserEntity.java]: Also, is there a specific reason for using an embedded ID for UserId?

Comment (hdprajwal) [reply mainframe/src/main/java/com/PiXl/mainframe/entities/UserEntity.java]: Nope, I should have removed it and used the other approadh was testing out something and forgot to change it later. Will update this.

Comment (ChiragBellara7) [reply mainframe/src/main/java/com/PiXl/mainframe/entities/UserEntity.java]: Alright. Other than this, everything looks fine to me.
I will resolve this conversation and approve the merge request from my end.

**Review by ChiragBellara7 — APPROVED** [review]

File: mainframe/src/main/java/com/PiXl/mainframe/entities/UserEntity.java (hunk 1 above)
Comment (hdprajwal) [reply mainframe/src/main/java/com/PiXl/mainframe/entities/UserEntity.java]: I have resolved this issue. will try to keep in mind to revert my testing changes before pushing next time.

**Pull request merged**